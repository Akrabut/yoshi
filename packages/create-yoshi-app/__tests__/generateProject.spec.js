const tempy = require('tempy');
const getProjectTypes = require('../src/getProjectTypes');
const { generateProject } = require('../src');
const getFilesInDir = require('../src/getFilesInDir');
const Answers = require('../src/Answers');

const projectTypes = getProjectTypes();

const mockedAnswers = projectTypes.map(projectType => [
  projectType,
  new Answers({
    projectName: `test-${projectType}`,
    authorName: 'rany',
    authorEmail: 'rany@wix.com',
    organization: 'wix',
    projectType: projectType.replace('-typescript', ''),
    transpiler: projectType.endsWith('typescript') ? 'typescript' : 'babel',
  }),
]);

test.each(mockedAnswers)('%s', (testName, answers) => {
  const tempDir = tempy.directory();

  generateProject(answers, tempDir);

  const files = getFilesInDir(tempDir);

  expect(files).toMatchSnapshot();
});
