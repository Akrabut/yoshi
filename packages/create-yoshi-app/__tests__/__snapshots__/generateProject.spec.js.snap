// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify generation works as expected 1`] = `
Object {
  ".gitignore": "",
  "TEST_PROJECT.json": "",
  "no-template.js": "console.log('foo');
",
  "package.json": "",
  "template.txt": "test-project
rany
rany@wix.com
wix

TEST_PROJECT
RANY
RANY_WIX_COM
WIX
",
}
`;

exports[`verify generation works as expected with extended template 1`] = `
Object {
  "bar.js": "console.log('bar');
",
  "foo.js": "console.log('foo');
",
}
`;
